CONTRIBUTING.md

```diff
@@ -82,13 +82,20 @@ changes.
 
 ### Developing and Running SQLFluff Locally
 
+#### Requirements
+
 The simplest way to set up a development environment is to use `tox`.
-First ensure that you have tox installed (windows users may have to replace `python3` with `py`):
+First ensure that you have tox installed:
 ```shell
-python3 -m pip install -U tox
+python3.8 -m pip install -U tox
 ```
+**IMPORTANT:** `tox` must be installed using Python 3.8 or newer. Otherwise `mypy` checks may fail. Depending on your setup and personal preference, replace `python3.8` as necessary, but make sure that the python version is new enough.
+
+<!-- Hopefully when tox 4 is released it will be possible to set `basepython = python3.10, python3.9, python3.8` to enforce a working version for running mypy checks. For now it's only possible to set it to a single version which is too restrictive. On the other hand for now there is a requirement for contributors to install tox with a new enough Python version. -->
```

Comment:
I think this is a little longwinded and potentially confusing. I note you commented it out, but I for one often read CONTRIBUTING.md and the like in my IDE where it won’t be hidden.

WDYT about this?:

```suggestion
Note: Unfortunately tox does not currently support setting just a minimum Python version (though this may be be coming in tox 4!).
```

Reply:
That's well put. Thanks.